crates/vault-registry/src/lib.rs

```diff
@@ -325,37 +289,235 @@ decl_module! {
+
+    /// Determines the rate for the collateral rate of Vaults,
+    /// at which users receive a premium, allocated from the
+    /// Vault's collateral, when performing a redeem with this Vault.
+    /// Must to be strictly greater than 100000 and LiquidationCollateralThreshold.
+    #[pallet::storage]
+    #[pallet::getter(fn premium_redeem_threshold)]
+    pub(super) type PremiumRedeemThreshold<T: Config> = StorageValue<_, UnsignedFixedPoint<T>, ValueQuery>;
+
+    /// Determines the lower bound for the collateral rate in issued tokens.
+    /// Must be strictly greater than 100000. If a Vault’s collateral rate
+    /// drops below this, automatic liquidation (forced Redeem) is triggered.
+    #[pallet::storage]
+    #[pallet::getter(fn liquidation_collateral_threshold)]
+    pub(super) type LiquidationCollateralThreshold<T: Config> = StorageValue<_, UnsignedFixedPoint<T>, ValueQuery>;
+
+    /// Account identifier of an artificial Vault maintained by the VaultRegistry
+    /// to handle issued balances and collateral of liquidated Vaults.
+    /// That is, when a Vault is liquidated, its balances are transferred to
+    /// LiquidationVault and claims are later handled via the LiquidationVault.
+    #[pallet::storage]
+    pub(super) type LiquidationVaultAccountId<T: Config> = StorageValue<_, T::AccountId, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn liquidation_vault)]
+    pub(super) type LiquidationVault<T: Config> = StorageValue<_, SystemVault<Issuing<T>>, ValueQuery>;
+
+    /// Mapping of Vaults, using the respective Vault account identifier as key.
+    #[pallet::storage]
+    pub(super) type Vaults<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId,
+        Vault<T::AccountId, T::BlockNumber, Issuing<T>, Backing<T>>,
+        ValueQuery,
+    >;
+
+    /// Mapping of reserved BTC addresses to the registered account
+    #[pallet::storage]
+    pub(super) type ReservedAddresses<T: Config> =
+        StorageMap<_, Blake2_128Concat, BtcAddress, T::AccountId, ValueQuery>;
+
+    /// Total collateral used for backing tokens issued by active vaults, excluding the liquidation vault
+    #[pallet::storage]
+    pub(super) type TotalUserVaultBackingCollateral<T: Config> = StorageValue<_, Backing<T>, ValueQuery>;
+
+    #[pallet::type_value]
+    pub(super) fn DefaultForStorageVersion() -> Version {
+        Version::V0
+    }
+
+    /// Build storage at V1 (requires default 0).
+    #[pallet::storage]
+    #[pallet::getter(fn storage_version)]
+    pub(super) type StorageVersion<T: Config> = StorageValue<_, Version, ValueQuery, DefaultForStorageVersion>;
+
+    /// Map of Vault Operators
+    #[pallet::storage]
+    pub(super) type IsNominationOperator<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, bool, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        #[doc = " The minimum collateral (e.g. DOT/KSM) a Vault needs to provide"]
```

Comment:
Same as on `btc-relay` conversion, is this structure (`#[doc = ""]`) required or can we just use doc comments?

Reply:
This was auto-generated, but you are right, regular doc comments work too, I'll change